bazel/foreign_cc/nghttp2.patch

```diff
@@ -1,12 +1,29 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index e70d2d49..f9c5340e 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -273,7 +273,11 @@ check_type_size("ssize_t" SIZEOF_SSIZE_T)
+ if(SIZEOF_SSIZE_T STREQUAL "")
+   # ssize_t is a signed type in POSIX storing at least -1.
+   # Set it to "int" to match the behavior of AC_TYPE_SSIZE_T (autotools).
+-  set(ssize_t int)
++  if(WIN32 AND CMAKE_SIZEOF_VOID_P EQUAL 8)
++    set(ssize_t int64_t)
++  else()
++    set(ssize_t int)
++  endif()
+ endif()
+ # AC_TYPE_UINT8_T
+ # AC_TYPE_UINT16_T
```

Comment:
What are the plans to upstream this? As it stands, having this patch will make version bumps hard as we will need to treat the patch manually.

Reply:
A previous attempt to fix the `ssize_t` issue on Windows: https://github.com/nghttp2/nghttp2/issues/616 was reverted, since adding the typedef could break projects that have code that also `#define ssize_t` on Windows.

For the rest of the patch, there's been no movement on the PR we submitted back in July

Reply:
Thanks for the update. It would be great if we can get this Windows support landed upstream, CC @tatsuhiro-t. Do we at least have a roadmap for how to do this?

Reply:
How about adding a cmake flag to define ssize_t as int64_t?  By default, ssize_t is int for backward compatibility.